### PR TITLE
Numpy compatible diagflat operator

### PR DESCRIPTION
**Pull request by endvroy**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by roywei** [comment]
@mxnet-label-bot add [numpy]

File: src/operator/numpy/np_diagflat_op-inl.h [hunk 1]
```diff
@@ -0,0 +1,201 @@
+/*
```
Comment (haojin2) [inline src/operator/numpy/np_diagflat_op-inl.h]: Better put this op in `np_init_op` files.

Comment (haojin2) [reply src/operator/numpy/np_diagflat_op-inl.h]: Seems like there's no `np_init_op-inl.h`, you could probably rename this `-inl.h` to that.

File: tests/python/unittest/test_numpy_op.py [hunk 2]
```diff
@@ -28,6 +28,59 @@
...
+                    mx_out.backward()
+                    assert_almost_equal(x.grad.asnumpy(), _np.ones(x.shape),
+                                        rtol=1e-3, atol=1e-5)
+
+                    # test imperative once again
+                    mx_out = np.diagflat(x, k)
+                    np_out = _np.diagflat(x.asnumpy(), k)
+                    assert_almost_equal(mx_out.asnumpy(), np_out, rtol=rtol,
+                                        atol=atol)
+
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: 2 blank lines between functions.

File: src/operator/numpy/np_diagflat_op.cu [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file np_diagflat_op.cu
+ * \brief CPU Implementation of numpy-compatible diagflat operator
```
Comment (haojin2) [inline src/operator/numpy/np_diagflat_op.cu]: `GPU Implementation`?
Actually please move those registrations to `np_init_op` files.

File: tests/python/unittest/test_numpy_op.py (hunk 2 above)
Comment (endvroy) [reply tests/python/unittest/test_numpy_op.py]: fixed

File: src/operator/numpy/np_diagflat_op.cu (hunk 3 above)
Comment (endvroy) [reply src/operator/numpy/np_diagflat_op.cu]: moved

File: src/operator/numpy/np_diagflat_op-inl.h (hunk 1 above)
Comment (endvroy) [reply src/operator/numpy/np_diagflat_op-inl.h]: done

**Comment by haojin2** [comment]
`diagflat` has been added in #16813, closing this PR.

**Pull request closed without merging**